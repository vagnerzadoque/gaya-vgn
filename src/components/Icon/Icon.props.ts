import { IconName } from '@naturacosmeticos/natds-icons'
import { Color, Size } from '@naturacosmeticos/natds-themes'
import { BrandTypes } from '../../brandTypes/brandTypes'

export type IconSize = keyof Size
export type IconColor = keyof Color
export interface IconProps {
  /**
   * Removes the entire element from the accessibility tree
   * when the Icon is purely decorative
   * @default true
   */
  ariaHidden?: boolean

  brand?: BrandTypes

  /**
   * Use to define a string that labels the current element
   * if your icon is not decorative
   */
  ariaLabel?: string,

  /**
  * The Icon color
   * @default "primary"
   */
  color?: IconColor;

  /**
   * Override or extend the styles applied to the component
   */
  className?: string

  /**
   * Set the icon to be rendered.
   * Check all available names in [Icon Library](https://ds.natura.design/28db352be/p/94367e-icon-library/b/6154b9)
   * @default outlined-default-mockup
   */
  name: IconName;

  /**
   * @default 'img'
   */
  role?: 'img' | 'button',

  /**
  * The Icon size
   * @default "standard"
   */
  size?: IconSize;

  /**
   * Optional ID for testing
   */
  testID?: string,
}
