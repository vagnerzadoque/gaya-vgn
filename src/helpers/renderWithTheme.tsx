import React from 'react'
import { JssProvider, SheetsRegistry } from 'react-jss'
import themes from '@naturacosmeticos/natds-themes'
import { render } from '@testing-library/react'
import { ThemeProvider } from '../ThemeProvider'

const renderWithTheme = (component: JSX.Element) => {
  const sheets = new SheetsRegistry()

  const renderTree = render(
    <JssProvider registry={sheets}>
      <ThemeProvider theme={themes.natura.light}>
        {component}
      </ThemeProvider>
    </JssProvider>
  )

  return { styles: sheets.toString(), component: renderTree }
}

export default renderWithTheme
