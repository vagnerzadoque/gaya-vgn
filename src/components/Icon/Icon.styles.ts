// lib icon version 1.27.5
import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { IconProps } from './Icon.props'
import { buildTheme } from '../../ThemeProvider'

type IconStyleProps = Pick<IconProps, 'size' | 'color' | 'brand' >

const getColorThemeIcon = (theme: Theme) => ({ brand, color }: IconStyleProps) => {
  const themeSelectIcon = buildTheme(brand, 'light')
  if (brand) {
    return color && themeSelectIcon.color[color]
  }
  return color && theme.color[color]
}

const styles = createUseStyles((theme: Theme) => ({
  icon: {
    color: ({ color, brand }: IconStyleProps) => getColorThemeIcon(theme)({ brand, color }),
    fontFamily: 'natds-icons',
    fontSize: ({ size }: IconStyleProps) => size && theme.size[size],
    pointerEvents: 'none',
    userSelect: 'none'
  }
}))

export default styles
