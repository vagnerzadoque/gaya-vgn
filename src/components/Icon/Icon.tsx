import React from 'react'
import { icons } from '@naturacosmeticos/natds-icons'
import { IconProps } from './Icon.props'
import styles from './Icon.styles'

export const checkIcon = (iconName: string): string => (iconName in icons ? iconName : 'outlined-default-mockup')

const Icon = React.forwardRef<HTMLElement, IconProps>(({
  ariaHidden = true,
  ariaLabel,
  className = '',
  color = 'highlight',
  brand,
  name,
  role = 'img',
  size = 'standard',
  testID = `icon-${name}`,
  ...props
}, ref) => {
  const { icon } = styles({ size, color, brand })
  const defaultIconName = `natds-icons-${checkIcon('outlined-default-mockup')}`
  const iconName = name ? `natds-icons-${checkIcon(name)}` : defaultIconName

  return (
    <i
      aria-hidden={ariaHidden}
      aria-label={ariaLabel}
      className={`${className} ${icon} natds-icons ${iconName} Icon-gaya`}
      data-testid={testID}
      ref={ref}
      role={role}
      {...props}
    />
  )
})

export default Icon
