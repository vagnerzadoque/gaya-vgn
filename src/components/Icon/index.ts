import Icon from './Icon'

export { Icon }
export { IconProps } from './Icon.props'
export default Icon
