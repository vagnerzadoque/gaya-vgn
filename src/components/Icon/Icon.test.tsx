import * as React from 'react'
import Icon from '.'
import renderWithTheme from '../../helpers/renderWithTheme'
import { checkIcon } from './Icon'

describe('Icon component', () => {
  it('should render correctly with default props', () => {
    const { styles, component: { getByTestId } } = renderWithTheme(<Icon name="outlined-default-mockup" />)
    const component = getByTestId(/mockup/)

    expect(component).toHaveStyle({ fontSize: 24, color: '#111111' })
    expect([styles.toString(), component]).toMatchSnapshot()
  })
  it('should render correctly with `medium` size', () => {
    const { styles, component: { getByTestId } } = renderWithTheme(<Icon name="outlined-default-mockup" size="medium" />)
    const component = getByTestId(/mockup/)

    expect(component).toHaveStyle({ fontSize: 48 })
    expect([styles.toString(), component]).toMatchSnapshot()
  })
  it('should render correctly with `primary` color', () => {
    const { styles, component: { getByTestId } } = renderWithTheme(<Icon name="outlined-default-mockup" color="primary" />)
    const component = getByTestId(/mockup/)

    expect(component).toHaveStyle({ color: '#f4ab34' })
    expect([styles.toString(), component]).toMatchSnapshot()
  })
})
describe('CheckIcon', () => {
  it('should return the default icon name', () => {
    const checkedIcon = checkIcon('unexpected-icon-name')

    expect(checkedIcon).toBe('outlined-default-mockup')
  })
  it('should return the given icon name', () => {
    const iconChecked = checkIcon('outlined-social-contact')

    expect(iconChecked).toBe('outlined-social-contact')
  })
})
