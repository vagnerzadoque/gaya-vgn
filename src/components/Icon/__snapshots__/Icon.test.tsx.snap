// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should render correctly with \`medium\` size 1`] = `
[
  ".icon-0-2-1 {
  font-family: natds-icons;
  user-select: none;
  pointer-events: none;
}
.icon-d0-0-2-2 {
  color: #111111;
  font-size: 48px;
}",
  <i
    aria-hidden="true"
    class=" icon-0-2-1 icon-d0-0-2-2 natds-icons natds-icons-outlined-default-mockup Icon-gaya"
    data-testid="icon-outlined-default-mockup"
    role="img"
  />,
]
`;

exports[`Icon component should render correctly with \`primary\` color 1`] = `
[
  ".icon-0-2-1 {
  font-family: natds-icons;
  user-select: none;
  pointer-events: none;
}
.icon-d0-0-2-2 {
  color: #f4ab34;
  font-size: 24px;
}",
  <i
    aria-hidden="true"
    class=" icon-0-2-1 icon-d0-0-2-2 natds-icons natds-icons-outlined-default-mockup Icon-gaya"
    data-testid="icon-outlined-default-mockup"
    role="img"
  />,
]
`;

exports[`Icon component should render correctly with default props 1`] = `
[
  ".icon-0-2-1 {
  font-family: natds-icons;
  user-select: none;
  pointer-events: none;
}
.icon-d0-0-2-2 {
  color: #111111;
  font-size: 24px;
}",
  <i
    aria-hidden="true"
    class=" icon-0-2-1 icon-d0-0-2-2 natds-icons natds-icons-outlined-default-mockup Icon-gaya"
    data-testid="icon-outlined-default-mockup"
    role="img"
  />,
]
`;
